native/src/main/java/io/ballerina/stdlib/xmldata/XmlToJson.java

```diff
@@ -202,13 +202,15 @@ private static void processAttributeWithAnnotation(BXmlItem xmlItem, String attr
                 } else if (fieldType instanceof UnionType) {
                     UnionType bUnionType = (UnionType) fieldType;
                     for (Type memberType : bUnionType.getMemberTypes()) {
-                        if (memberType.getTag() == TypeTags.RECORD_TYPE_TAG)  {
+                        if (memberType.getTag() == TypeTags.TYPE_REFERENCED_TYPE_TAG)  {
                             annotationType = memberType;
                         }
                     }
                 }
                 if (annotationType.getTag() == TypeTags.RECORD_TYPE_TAG) {
                     annotations = ((RecordType) annotationType).getAnnotations();
+                } else if (annotationType.getTag() == TypeTags.TYPE_REFERENCED_TYPE_TAG) {
+                    annotations = ((RecordType) TypeUtils.getReferredType(annotationType)).getAnnotations();
```

Comment:
we can remove this `else if` part if we use `TypeUtils.getReferredType` earlier as per the previous suggestion.